From f5125725e4e2e250ccc78a17a8b77431100e7c15 Mon Sep 17 00:00:00 2001
From: Armin kuster <akuster808@gmail.com>
Date: Wed, 14 Oct 2020 22:48:31 -0700
Subject: [PATCH] Busybox does not support ps -p so use pgrep

Upstream-Status: Inappropriate [embedded specific]
Based on changes from Diego Sueiro <Diego.Sueiro@arm.com>

Signed-off-by: Armin kuster <akuster808@gmail.com>

Refresh to apply on top of 2.6.1.

Signed-off-by: Trevor Gamblin <tgamblin@baylibre.com>
---
 src/bin/keactrl/keactrl.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/bin/keactrl/keactrl.in b/src/bin/keactrl/keactrl.in
index cccfdac303..20ae2e6ec5 100644
--- a/src/bin/keactrl/keactrl.in
+++ b/src/bin/keactrl/keactrl.in
@@ -146,8 +146,8 @@ check_running() {
     # Get the PID from the PID file (if it exists)
     get_pid_from_file "${proc_name}"
     if [ ${_pid} -gt 0 ]; then
-        # Use ps to check if PID is alive
-        if ps -p ${_pid} 1>/dev/null; then
+        # Use pgrep and grep to check if PID is alive
+        if pgrep -v 1 | grep ${_pid} 1>/dev/null; then
             # No error, so PID IS ALIVE
             _running=1
         fi
-- 
2.39.2

